### PR TITLE
docs(guides/data-loading | api/conventions): fix type inference & add json import

### PR DESCRIPTION
**Pull request by hakimLyon**

### CODE REVIEW

File: docs/guides/data-loading.md [hunk 1]
```diff
@@ -186,14 +186,12 @@ export default function ProductCategory() {
 If you are using TypeScript, you can use type inference to use Prisma Client generated types on when calling `useLoaderData`. This allows better type safety and intellisense when writing your code that uses the loaded data.
 
 ```tsx filename=tsx filename=app/routes/products/$productId.tsx
-import type { LoaderFunction } from "@remix-run/node"; // or cloudflare/deno
+import type { LoaderArgs } from '@remix-run/node'; // or cloudflare/deno
```
Comment (machour) [inline docs/guides/data-loading.md:189]:
```suggestion
import type { LoaderArgs } from "@remix-run/node"; // or cloudflare/deno
```

File: docs/guides/data-loading.md [hunk 2]
```diff
@@ -209,19 +207,18 @@ async function getLoaderData(productId: string) {
...
-  params,
-}) => {
-  return json<LoaderData>(
-    await getLoaderData(params.productId)
-  );
+export const loader = async ({ params }: LoaderArgs) => {
+  let LoaderData: Awaited<ReturnType<typeof getLoaderData>> | null = null
+  LoaderData = await getLoaderData(`${params.productId}`)
+
+  return json({product: LoaderData});
```
Comment (machour) [inline docs/guides/data-loading.md:214]: is this all needed? could you try:

```ts
export const loader = async ({ params }: LoaderArgs) => {
  const product = await getLoaderData(params.productId);

  return json({  product });
```

and see if type inference works?

**Review by machour — CHANGES_REQUESTED** [review]
Thank you @hakimLyon for this update.
I think the code could be improved, can you have a look at my comment?

File: docs/guides/data-loading.md (hunk 2 above)
Comment (hakimLyon) [reply docs/guides/data-loading.md:214]: yea thanks it's better. type inference work.

File: docs/guides/data-loading.md [hunk 3]
```diff
@@ -209,19 +207,17 @@ async function getLoaderData(productId: string) {
...
 }
 
-export const loader: LoaderFunction = async ({
-  params,
-}) => {
-  return json<LoaderData>(
-    await getLoaderData(params.productId)
-  );
+export const loader = async ({ params }: LoaderArgs) => {
+ const product = await getLoaderData(params.productId)
```
Comment (machour) [inline docs/guides/data-loading.md:211]:
```suggestion
  const product = await getLoaderData(params.productId);
```

🤓 🙈

**Review by machour — APPROVED** [review]

File: docs/guides/data-loading.md (hunk 3 above)
Comment (hakimLyon) [reply docs/guides/data-loading.md]: 🙈

File: docs/guides/data-loading.md [hunk 4]
```diff
@@ -209,19 +207,17 @@ async function getLoaderData(productId: string) {
...
 }
 
-export const loader: LoaderFunction = async ({
-  params,
-}) => {
-  return json<LoaderData>(
-    await getLoaderData(params.productId)
-  );
+export const loader = async ({ params }: LoaderArgs) => {
+ const product = await getLoaderData(params.productId);
```
Comment (machour) [inline docs/guides/data-loading.md:211]:
```suggestion
  const product = await getLoaderData(params.productId);
```

**Pull request closed without merging**